### PR TITLE
Add the functionality of plans

### PR DESCRIPTION
**Pull request by ZeeshanAzhar23**

### CODE REVIEW

File: app/controllers/plans_controller.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+class PlansController < ApplicationController
+  before_action :find_plan, only: [:show,:edit,:update,:destroy]
```
Comment (javier459) [inline app/controllers/plans_controller.rb]: spacing issue

File: app/controllers/plans_controller.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  around_action :delete_plan, only: [:destroy]
+  def index
+  @plans=Plan.all
```
Comment (javier459) [inline app/controllers/plans_controller.rb]: spacing issue

File: app/controllers/plans_controller.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  authorize @plans
+  end
+
+  def new
+    @plan=Plan.new
```
Comment (javier459) [inline app/controllers/plans_controller.rb]: spacing issue

File: app/controllers/plans_controller.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    authorize @plan
+  end
+  def create
+    @plan=Plan.create(plan_params)
```
Comment (javier459) [inline app/controllers/plans_controller.rb]: spacing issue

File: app/controllers/plans_controller.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    authorize @plan
+    if @plan.update(plan_params)
+      redirect_to plan_path(@plan)
+    else
+      render 'edit'
+    end
+  end
+
+  def show
+    @features = Plan.find(params[:id]).features.new
```
Comment (javier459) [inline app/controllers/plans_controller.rb]: we already have @plan accessible here, we can use `@plan.features.new` here

File: app/controllers/plans_controller.rb [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  private def find_plan
+    @plan = Plan.find(params[:id])
+  end
+  private def delete_plan
+    @stripe_prod_id=@plan.stripe_plan_id
+    @stripe_price_id=@plan.price_id
+    yield
+    #Strip::Price.delete(@stripe_price_id)
+    Stripe::Product.update(@stripe_prod_id,active: 'false')
+  end
```
Comment (javier459) [inline app/controllers/plans_controller.rb]: add private block and then use these methods

File: app/controllers/plans_controller.rb [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+  private def find_plan
+    @plan = Plan.find(params[:id])
+  end
+  private def delete_plan
+    @stripe_prod_id=@plan.stripe_plan_id
+    @stripe_price_id=@plan.price_id
+    yield
+    #Strip::Price.delete(@stripe_price_id)
```
Comment (javier459) [inline app/controllers/plans_controller.rb]: remove commented code

File: app/models/plan.rb [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    update(stripe_plan_id: @plan.id)
+    @price=Stripe::Price.create({
+      product: @plan.id,
+      unit_amount: monthly_fee.to_i,
+      currency:'usd',
+      recurring: {interval: 'month'},
+      metadata: {order_id: '6735'},
+    })
+    update(price_id: @price.id)
+  end
```
Comment (javier459) [inline app/models/plan.rb]: create a separate function for this and call that function in callback

File: app/models/plan.rb [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+  # def return_price_id
+  #   puts 'chcking'
+  #   @price.id
+  #   puts @price.id
+  # end
```
Comment (javier459) [inline app/models/plan.rb]: remove commented code

File: app/views/plans/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+      <%# <td><%= plan.plan.email.split('@')[0].capitalize %>
+        <td><%= plan.monthly_fee %></td>
+        <td><%= link_to 'Show', plan_path(plan) %></td>
+        <%if current_user && current_user.role == 'admin'%>
+        <td><%= link_to 'Edit', edit_plan_path(plan) %></td>
+        <td><%= link_to 'Destroy', plan, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+        <%end %>
+      </tr>
+      <%count = count.next%>
+    <% end %>
```
Comment (javier459) [inline app/views/plans/index.html.erb]: use partial here to remove the loop.

File: app/views/users/index.html.erb [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+      <%# <><%= user.user.email.split('@')[0].capitalize %>
+        <td><%= user.role %></td>
+        <td><%= link_to 'Show', user_path(user) %></td>
+        <%if current_user && current_user.role == 'admin'%>
+        <td><%= link_to 'Edit', edit_user_path(user) %></td>
+        <td><%= link_to 'Destroy', user, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+        <%end %>
+      </tr>
+      <%count = count.next%>
+    <% end %>
```
Comment (javier459) [inline app/views/users/index.html.erb]: use partial here

File: app/views/users/old_code.html.erb [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+    <div>
+      <%= button_to 'Manage Billing', billings_path ,class: 'btn btn-primary' %>
+    </div>
+  <% end %>
+</div>
+
+<%# <% if current_user.admin?%>
+<%# <h3>Cancel account</h3> %>
+<%# <p>Unhappy? <%= button_to "Cancel account", registration_path(@user_name), data: { confirm: "Are you sure?" }, method: :delete,class: 'btn btn-primary btn-danger'%></p>
+<%# <%end%>
```
Comment (javier459) [inline app/views/users/old_code.html.erb]: remove commented code.

**Review by javier459 — CHANGES_REQUESTED** [review]

**Review by ZeeshanAzhar23 — COMMENTED** [review]
Kindly Review the changes made

**Review by javier459 — APPROVED** [review]

**Pull request merged**